### PR TITLE
system-config-printer: 1.5.11 -> 1.5.12

### PR DESCRIPTION
**Pull request by symphorien**
Fixes some dialogs with our version of gtk like 
```
 grid.attach (entry, 1, 2, i, i + 1, 0, 0)
TypeError: Gtk.Grid.attach() takes exactly 6 arguments (8 given)
```

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

**Comment by teto** [comment]
@GrahamcOfBorg build system-config-printer

**Review by worldofpeace — CHANGES_REQUESTED** [review]
Can we update the `homepage` and document the source url change (in commit message)?
And what purpose can adding packagekit support be in nixos? I'm not sure it looks like it needs it as a buildtime dependency 
* https://github.com/OpenPrinting/system-config-printer/blob/4fd9952fe83757a8592d4e07848367930a4888db/ppdsloader.py#L216

**Comment by symphorien** [comment]
* `packagekit` is a runtime dependency `$out/bin/install-printerdriver`.
* The previous url (https://github.com/zdohnal/system-config-printer) is a fork of the new one, according to github.

**Comment by symphorien** [comment]
Also fedora did this as well:  (see changelog for 1.5.11-17):
https://apps.fedoraproject.org/packages/system-config-printer/sources/spec/

I changed the homepage in meta and enabled tests.

**Comment by worldofpeace** [comment]
> 1.5.11-17

I wonder why fedora doesn't declare any dependence on `packagekit`.

> packagekit is a runtime dependency $out/bin/install-printerdriver.

So then I guess the `system-config-printer` nixos module/s needs to enable `services.packagekit.enable` then. And yes it does need it wrapped with the typelibs
* https://github.com/OpenPrinting/system-config-printer/blob/master/install-printerdriver.py#L5

File: pkgs/tools/misc/system-config-printer/default.nix [hunk 1]
```diff
@@ -54,20 +63,14 @@ stdenv.mkDerivation rec {
...
-          -i "$out/etc/systemd/system/configure-printer@.service"
+      patchPythonScript $out/etc/udev/udev-add-printer
 
       substituteInPlace $out/etc/udev/rules.d/70-printers.rules \
         --replace "udev-configure-printer" "$out/etc/udev/udev-configure-printer"
     '';
 
   meta = {
-    homepage = http://cyberelk.net/tim/software/system-config-printer/;
+    inherit (src.meta) homepage;
```
Comment (worldofpeace) [inline pkgs/tools/misc/system-config-printer/default.nix]: I'd prefer not to do this. I think generally it's useful to look at an expression and actually be able to search the homepage link.

Comment (worldofpeace) [reply pkgs/tools/misc/system-config-printer/default.nix]: Cosmetically we could probably do `with stdenv.lib;` here as well.

Comment (symphorien) [reply pkgs/tools/misc/system-config-printer/default.nix]: I replaced this by the literal url.

**Comment by symphorien** [comment]
Modified the dependency on packagekit in the module (whose existence I discovered with your comment).

**Review by worldofpeace — CHANGES_REQUESTED** [review]
Can we default to python3?

**Comment by symphorien** [comment]
We already use python3: https://github.com/NixOS/nixpkgs/blob/master/pkgs/top-level/all-packages.nix#L6423

**Comment by worldofpeace** [comment]
> We already use python3: https://github.com/NixOS/nixpkgs/blob/master/pkgs/top-level/all-packages.nix#L6423

I see. Can we get rid of that? I find it makes the expression hard to read, and overriding shouldn't be any different.

**Comment by symphorien** [comment]
done

**Comment by symphorien** [comment]
Is there anything that needs to be amended yet ?

**Comment by worldofpeace** [comment]
Thanks @symphorien.

**Pull request merged**